test/testOnly/controllers/PortalStubControllerSpec.scala

```diff
@@ -41,17 +42,20 @@ class PortalStubControllerSpec extends ControllerBaseSpec {
 
     val mockEnrolmentsAuthService: EnrolmentsAuthService = new EnrolmentsAuthService(mockAuthConnector)
     val mockAccountDetailsService = mock[AccountDetailsService]
-    val mockHybridUserPredicate: HybridUserPredicate = new HybridUserPredicate(mockAccountDetailsService, mockServiceErrorHandler)
-    val mockAgentPredicate: AgentPredicate = new AgentPredicate(mockEnrolmentsAuthService, messages, mockAppConfig)
+    val mockHybridUserPredicate: HybridUserPredicate = new HybridUserPredicate(mockAccountDetailsService, mockServiceErrorHandler, mcc, ec)
```

Comment:
There's a bunch of stuff being created in this trait that already exists in `ControllerBaseSpec`. Can you re-use those vals where possible?